File: myapplication_ui/src/services/DriverService.js
import http from "../http-common"; 

class DriverService {
  getAllDrivers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/driver/drivers`, searchDTO);
  }

  get(driverId) {
    return this.getRequest(`/driver/${driverId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/driver?field=${matchData}`, null);
  }

  addDriver(data) {
    return http.post("/driver/addDriver", data);
  }

  update(data) {
  	return http.post("/driver/updateDriver", data);
  }
  
  uploadImage(data,driverId) {
  	return http.postForm("/driver/uploadImage/"+driverId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new DriverService();

File: src/main/java/com/myapplication/service/DriverService.java
package com.myapplication.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myapplication.domain.Driver;
import com.myapplication.dto.DriverDTO;
import com.myapplication.dto.DriverSearchDTO;
import com.myapplication.dto.DriverPageDTO;
import com.myapplication.dto.DriverConvertCriteriaDTO;
import com.myapplication.service.GenericService;
import com.myapplication.dto.common.RequestDTO;
import com.myapplication.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface DriverService extends GenericService<Driver, Integer> {

	List<Driver> findAll();

	ResultDTO addDriver(DriverDTO driverDTO, RequestDTO requestDTO);

	ResultDTO updateDriver(DriverDTO driverDTO, RequestDTO requestDTO);

    Page<Driver> getAllDrivers(Pageable pageable);

    Page<Driver> getAllDrivers(Specification<Driver> spec, Pageable pageable);

	ResponseEntity<DriverPageDTO> getDrivers(DriverSearchDTO driverSearchDTO);
	
	List<DriverDTO> convertDriversToDriverDTOs(List<Driver> drivers, DriverConvertCriteriaDTO convertCriteria);

	DriverDTO getDriverDTOById(Integer driverId);







}





